src/Mvc/Mvc.Core/src/Routing/ControllerRequestDelegateFactory.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    internal class ControllerRequestDelegateFactory : IRequestDelegateFactory
```

Comment:
Have you tried implementing this yet for Pages? I suspect it's more complex

Reply:
I have a stash for pages and it is more complex.